src/diffusers/pipelines/audioldm2/pipeline_audioldm2.py

```diff
@@ -368,16 +407,24 @@ def encode_prompt(
 
         # Define tokenizers and text encoders
         tokenizers = [self.tokenizer, self.tokenizer_2]
-        text_encoders = [self.text_encoder, self.text_encoder_2]
+
+        if self.text_encoder_2.config.model_type == "vits":
```

Comment:
We can add variables here `is_vits_text_encoder = isinstance(self.text_encoder_2, VitsModel)` and use it to check everywhere in the pipeline.